--- pkg/helpers.js	2021-06-21 22:20:28.538393581 +0200
+++ pkg/helpers.js.new	2021-06-21 22:27:12.478124395 +0200
@@ -19,72 +19,18 @@
   }
 }
 
-// checks if being run on the windows subsystem for linux:
-// used to modify windows notion app.
-const is_wsl =
-    process.platform === 'linux' &&
-    os.release().toLowerCase().includes('microsoft'),
-  // ~/.notion-enhancer absolute path.
-  __data = path.resolve(
-    `${
-      is_wsl
-        ? (() => {
-            const stdout = execSync('cmd.exe /c echo %systemdrive%%homepath%', {
-                encoding: 'utf8',
-              }),
-              drive = stdout[0];
-            return `/mnt/${drive.toLowerCase()}${stdout
-              .replace(/\\/g, '/')
-              .slice(2)
-              .trim()}`;
-          })()
-        : os.homedir()
-    }/.notion-enhancer`
-  );
+// patched: stripped unnecessary logic for WSL
+// ~/.notion-enhancer absolute path.
+const __data = `${os.homedir()}/.notion-enhancer`
 
-// transform a wsl filepath to its relative windows filepath if necessary.
+// patched: stripped unnecessary logic for WSL
 function realpath(hack_path) {
-  if (!is_wsl) return hack_path.replace(/\\/g, '/');
-  hack_path = fs.realpathSync(hack_path);
-  if (hack_path.startsWith('/mnt/')) {
-    hack_path = `${hack_path[5].toUpperCase()}:${hack_path.slice(6)}`;
-  } else hack_path = `//wsl$/${process.env.WSL_DISTRO_NAME}${hack_path}`;
-  return hack_path;
+  return hack_path.replace(/\\/g, '/');
 }
 
-// gets possible system notion app filepaths.
+// patched: stripped unnecessary logic for runtime path detection
 function getNotionResources() {
-  let folder = '';
-  switch (process.platform) {
-    case 'darwin':
-      folder = '/Applications/Notion.app/Contents/Resources';
-      break;
-    case 'win32':
-      folder = process.env.LOCALAPPDATA + '\\Programs\\Notion\\resources';
-      break;
-    case 'linux':
-      if (is_wsl) {
-        const stdout = execSync('cmd.exe /c echo %localappdata%', {
-            encoding: 'utf8',
-          }),
-          drive = stdout[0];
-        folder = `/mnt/${drive.toLowerCase()}${stdout
-          .replace(/\\/g, '/')
-          .slice(2)
-          .trim()}/Programs/Notion/resources`;
-      } else {
-        for (let loc of [
-          '/usr/lib/notion-desktop/resources', // https://github.com/davidbailey00/notion-deb-builder/
-          '/opt/notion-app', // https://aur.archlinux.org/packages/notion-app/
-          '/opt/notion', // https://github.com/jaredallard/notion-app
-        ]) {
-          if (fs.pathExistsSync(loc)) folder = loc;
-        }
-      }
-  }
-  if (!folder)
-    throw new EnhancerError('nothing found: platform not supported.');
-  return folder;
+  return path.resolve(__dirname, '../../../');
 }
 
 // lists/fetches all available extensions + themes
@@ -184,7 +130,6 @@
 
 module.exports = {
   EnhancerError,
-  is_wsl,
   __data,
   realpath,
   getNotionResources,
