--- mods/core/render.js	2021-08-15 22:49:21.087926244 +0200
+++ mods/core/render.new.js	2021-08-15 22:49:54.986617806 +0200
@@ -708,7 +708,7 @@
                   },
                   React.createElement('span', {
                     dangerouslySetInnerHTML: {
-                      __html: (title.img || '') + (title.text || 'notion.so'),
+                      __html: (title.img || '').replace('notion://', 'https://') + (title.text || 'notion.so'),
                     },
                   }),
                   React.createElement(
