--- main/main.js	2021-12-15 19:09:58.501121407 +0100
+++ main/main.js	2021-12-15 19:14:20.397000000 +0100
@@ -23,6 +23,7 @@
 };
 Object.defineProperty(exports, "__esModule", { value: true });
 const electron_1 = require("electron");
+electron_1.app.commandLine.appendSwitch("--no-sandbox");
 require("isomorphic-fetch");
 require("./crashReporter");
 const autoUpdater_1 = require("./autoUpdater");
