--- main/schemeHandler.js	2021-08-28 14:28:05.566193462 +0200
+++ main/schemeHandler.new.js	2021-08-28 14:44:37.106271190 +0200
@@ -122,6 +122,16 @@
 function registerUrlSchemeProxy() {
     const { protocol } = electron_1.session.fromPartition(constants_1.electronSessionPartition);
     const success = protocol.registerStreamProtocol(config_1.default.protocol, async (req, callback) => {
+        {
+          const notionEnhancerSchemePrefix = 'notion://www.notion.so/__notion-enhancer/';
+          if (req.url.startsWith(notionEnhancerSchemePrefix)) {
+            const notionEnhancerFilepath = `${__dirname}/../embedded_enhancer/mods/${req.url.slice(
+              notionEnhancerSchemePrefix.length,
+            )}`;
+            callback(require('fs').createReadStream(require('path').resolve(notionEnhancerFilepath)));
+          }
+        }
+        
         if (config_1.default.isLocalhost && !config_1.default.offline) {
             proxyRequest(req, callback);
             return;
